scipy/special/special/cephes/kolmogorov.h

```diff
@@ -0,0 +1,1043 @@
+ * [5] van Mulbregt, Paul,  "Computing the Cumulative Distribution Function and Quantiles of the limit of the Two-sided
+ Kolmogorov-Smirnov Statistic"
+ *     https://arxiv.org/abs/1803.00426
+ *
+ */
+
+#include "../config.h"
+#include "../error.h"
+
+#include "const.h"
+#include "dd_real.h"
+#include "unity.h"
+
+namespace special {
+namespace cephes {
+
+    namespace detail {
+        /* ************************************************************************ */
+        /* Algorithm Configuration */
+
+        constexpr int KOLMOG_MAXITER = 500;
+
+        /*
+         * Kolmogorov Two-sided:
+         * Switchover between the two series to compute K(x)
+         *  0 <=  x <= KOLMOG_CUTOVER and
+         *  KOLMOG_CUTOVER < x < infty
+         */
+        constexpr double KOLMOG_CUTOVER = 0.82;
+
+        /*
+         * Smirnov One-sided:
+         * n larger than SMIRNOV_MAX_COMPUTE_N will result in an approximation
+         */
+        constexpr int SMIRNOV_MAX_COMPUTE_N = 1000000;
+
+        /*
+         * Use the upper sum formula, if the number of terms is at most SM_UPPER_MAX_TERMS,
+         * and n is at least SM_UPPERSUM_MIN_N
+         * Don't use the upper sum if lots of terms are involved as the series alternates
+         *  sign and the terms get much bigger than 1.
+         */
+        constexpr int SM_UPPER_MAX_TERMS = 3;
+        constexpr int SM_UPPERSUM_MIN_N = 10;
+
+        /* ************************************************************************ */
+        /* ************************************************************************ */
+
+        /* exp() of anything below this returns 0 */
+        constexpr int MIN_EXPABLE = (-708 - 38);
+
+        /* Class to hold the CDF, SF and PDF, which are computed simultaneously */
+        class ThreeProbs {
+          public:
+            double sf;
+            double cdf;
+            double pdf;
+            ThreeProbs(double sf, double cdf, double pdf) : sf(sf), cdf(cdf), pdf(pdf) {}
+        };
+
+        constexpr double _xtol = std::numeric_limits<double>::epsilon();
+        constexpr double _rtol = 2 * _xtol;
+
+        SPECFUN_HOST_DEVICE inline bool _within_tol(double x, double y, double atol, double rtol) {
+            double diff = std::abs(x - y);
+            bool result = (diff <= (atol + rtol * std::abs(y)));
+            return result;
+        }
+
+        /* ************************************************************************ */
+        /* Kolmogorov : Two-sided                      **************************** */
+        /* ************************************************************************ */
+
+        SPECFUN_HOST_DEVICE inline ThreeProbs _kolmogorov(double x) {
+            double P = 1.0;
+            double D = 0;
+            double sf, cdf, pdf;
+
+            if (std::isnan(x)) {
+                return ThreeProbs(std::numeric_limits<double>::quiet_NaN(), std::numeric_limits<double>::quiet_NaN(),
```

Comment:
You can just use `{...}` here instead of explicitly calling `ThreeProbs` constructor.